segmentstore/server/host/src/main/java/io/pravega/segmentstore/server/host/handler/AppendProcessor.java

```diff
@@ -295,6 +318,39 @@ public void append(Append append) {
                 });
     }
 
+    private void updateSegmentEventCountCache(Append append) {
+        if (!isTransientSegment(append.getSegment()) && !isTransactionSegment(append.getSegment())) {
+            String eventCountIdentifier = (append.getSegment() + EVENT_COUNT_SUFFIX).trim();
+            WriterState state = this.writerStates.get(Pair.of(eventCountIdentifier, append.getWriterId()));
+            long previousEventCount = state.getEventSizeForAppend();
+            long newEventCount = previousEventCount + append.getEventCount();
+            this.writerStates.put(Pair.of(eventCountIdentifier, append.getWriterId()), new WriterState(newEventCount));
```

Comment:
This PR is very confused as to what the argument to `new WriterState(` is. In some places it is acting as event number, in others as event count, and elsewhere it is passes as `24` as though it were the size of the entries in the index.